### ISSUE TITLE
Thread not being marked as read

### ISSUE THREAD

**Issue by andybalaam** [vector-im/element-web#24442]
### Steps to reproduce

1. In a room with a thread
2. Open the thread and read it
3. Go to the all threads view - see the dot indicates that the thread is unread
4. Re-open the thread
5. Still not marked as read

#### What did you expect?

I expected the thread to be marked as read

#### What happened instead?

It was not marked as read

### Operating system

Ubuntu 22.04

### Browser information

Firefox 109.0.1

### URL for webapp

https://develop.element.io

### Application version

Element version: e047e5d803c4-react-a756b33fe912-js-2a363598dd98 Olm version: 3.2.12

### Homeserver

matrix.org

### Will you send logs?

Yes

**Comment by andybalaam** [comment]
Probably related to https://github.com/vector-im/element-web/issues/23685 but not the same because it persists after I open the room.

**Comment by weeman1337** [comment]
Also found a thread not being marked as read.
More about it in the rageshake.

**Comment by Walnussbaer** [comment]
We experience this problem as well in our company. But not ever time when someone posts something in a thread, just sometimes.

**Comment by MadLittleMods** [comment]
Reproduced this issue today ([rageshake logs](https://github.com/matrix-org/element-web-rageshakes/issues/20170))

Room list shows 1 unread message:

![](https://user-images.githubusercontent.com/558581/219527462-b313ad79-6373-44ed-83a5-df6bdd942559.png)

Thread list shows a dot:

![](https://user-images.githubusercontent.com/558581/219527476-61612c2e-dabf-416b-a209-e277c7f70284.png)

Clicking on the thread and scrolling up and down has no effect:

<img src="https://user-images.githubusercontent.com/558581/219527974-232254bd-6862-40ff-9d2d-dfe560b01722.png" width="150">


`/devtools` -> Notification debug output shows that I have read all of the messages in the thread. The last message is an edit to the thread parent.

Room status
 - Room unread status: Grey, count: 1
 - Notification state is all_messages
 - Room is not encrypted 🚨

Main timeline
 - Total: 0
 - Highlight: 0
 - Dot: false

Threads timeline
 - Thread Id: `$ANtMWAyX_RzChi195z7EHmQN6cs5g687RGlNQEzJeT4`
    - Total: 1
    - Highlight: 0
    - Dot: false
    - User read up to: `$nDgVIqgPXwtO2O1pQLdh6B2sZ_YrsS4I0rVfHt-i4I0`
    - Last event:
       - ID: `$nDgVIqgPXwtO2O1pQLdh6B2sZ_YrsS4I0rVfHt-i4I0`
       - Type: `m.room.message`
       - Sender: `@aw-bib:matrix.org`

**Comment by florianduros** [comment]
Reproduced in:

```
Room ID: !UCBgpJmTgAHYpcJBoh:matrix.org
Event ID: $7ZIMSb1MgmlVH_d1iEUAn9z-u3n_BN70AXHi1cRwDh4
Thread root ID: $hVLFDTGWWhDaqgku1JoX8uUrKcrZWF5DfC3N8vq-trw
```

Event

```
{
  "room_id": "!UCBgpJmTgAHYpcJBoh:matrix.org",
  "type": "m.room.message",
  "content": {
    "msgtype": "m.text",
    "body": ...
  }
}
```

**Comment by MadLittleMods** [comment]
@justjanne It feels really odd to add `X-Cannot-Reproduce` and `X-Needs-Info` for something that 4 Element employees (some on the Web team) have bothered to report. We've provided all information possible with the current tools. If more info is needed to reproduce, then we should add some better logging/tracing to the app for this so we can get you the info needed to reproduce locally yourself. I understand you technically can't reproduce yourself or at least consistently to figure out what's going wrong but it feels like that label is more for issues we're not really confident in.

**Comment by justjanne** [comment]
@MadLittleMods I labeled it cannot reproduce/needs info because I don't have enough info to reproduce it reliably or write a test to reproduce it.

Just because the labels are attached doesn't mean the issue is invalid — it definitely is valid, otherwise the issue wouldn't remain open and on the top of my todo list.

In fact, I've been spending pretty much my entire dev time in the past weeks on these issues.

Part of why the labels are attached (and why I'm not investigating further on this specific issue right now) is that I've discovered a solution that would solve and/or fix multiple issues at the same time, and I'd like to see how that solution affects this issue before I investigate further.
